chia/data_layer/data_store.py

```diff
@@ -655,3 +749,55 @@ async def get_proof_of_inclusion_by_key(
         async with self.db_wrapper.locked_transaction(lock=lock):
             node = await self.get_node_by_key(key=key, tree_id=tree_id, lock=False)
             return await self.get_proof_of_inclusion_by_hash(node_hash=node.hash, tree_id=tree_id, lock=False)
+
+    async def get_left_to_right_ordering(
+        self,
+        node_hash: bytes32,
+        tree_id: bytes32,
+        root_hash: bytes32,
+        *,
+        lock: bool = True,
+        num_nodes: int = 2500,
+    ) -> Tuple[bool, List[Dict[str, Any]]]:
+        ancestors = await self.get_ancestors_2(node_hash, tree_id, lock=True)
+        # Root hash changed, abort the process and start over again.
+        if root_hash != node_hash and root_hash != ancestors[-1].hash:
+            return (True, [])
+        stack = []
+        path_hashes = []
+        for ancestor in ancestors:
+            path_hashes.append(ancestor.hash)
+        for ancestor in reversed(ancestors):
+            assert isinstance(ancestor, InternalNode)
+            if ancestor.right_hash not in path_hashes and ancestor.right_hash != node_hash:
+                stack.append(ancestor.right_hash)
+        count = 0
+        nodes = []
+        while count < num_nodes:
+            count += 1
+            node = await self.get_node(node_hash)
+            if node is None:
+                return []
+            if isinstance(node, TerminalNode):
+                nodes.append(
+                    {
+                        "key": node.key.hex(),
+                        "value": node.value.hex(),
+                        "is_terminal": True,
+                    }
```

Comment:
This looks like an RPC-ready serialized form that the data store should not be aware of.